Docs: add a try Tetragon on Linux getting-started tutorial (#899)
cilium/tetragon: docs/content/en/docs/getting-started/try-tetragon-linux.md

- `quay.io/cilium/tetragon:{{< latest-version >}}` is the latest release of
  Tetragon available.

## Observe Tetragon base events

With this default configuration, Tetragon already loaded its base sensors to
perform [process lifecycle observability](/docs/tetragon-events/process-lifecycle/).

To quickly see the events, you can use the `tetra` CLI already shipped in the
Tetragon container that was just started, it will connect to the Tetragon gRPC
server listening on `localhost:54321`. In another terminal, type the following
command:

```shell
docker exec tetracontainer tetra getevents -o compact
```

In a different terminal, you can start a shell like `sh` and start executing
programs. Here, the commands executed in the shell where `ls`, `uname -a`,
`whoami`, and `exit`, the output should be similar to this:

```
🚀 process  /usr/bin/sh
🚀 process  /usr/bin/ls
💥 exit     /usr/bin/ls  0
🚀 process  /usr/bin/uname -a
💥 exit     /usr/bin/uname -a 0
🚀 process  /usr/bin/whoami
💥 exit     /usr/bin/whoami  0
💥 exit     /usr/bin/sh  0
```

You can see the start of `sh` first, then the execution of the multiple
programs, `ls`, `uname` and `whoami`, and finally the exit of `sh`.

{{< note >}}

> Since Tetragon is running directly on the host, you may observe a lot of
> unrelated from various process executions on your machine unrelated to what you
> typed during the shell session. Indeed, Tetragon is monitoring every process

Reviewer: ```suggestion
Since Tetragon monitors all processes on the host  host, you may observe events
unrelated to what was
typed in the shell session. Indeed, Tetragon is monitoring every process
```

start and exit in your environment.
{{< /note >}}

## Write a TracingPolicy to extend Tetragon events

> Tetragon observes the process lifecycle by default, but for more advanced use
> cases, you can write your own custom TracingPolicy. A TracingPolicy is a

Reviewer: ```suggestion
Tetragon by default observes only process lifecycles.  More advanced use-cases, 
require configuring Tetragon via TracingPolicies. A TracingPolicy is a
```

> user-configurable Kubernetes custom resource (CR) that allows users to trace
> arbitrary events in the kernel and optionally define actions to take on a
> match, for enforcement for example.

Reviewer: ```suggestion
user-configurable Kubernetes custom resource (CR) that allows users to access additional functionality,
such as tracing arbitrary events in the kernel and, optionally, define actions to take on a match.
These actions can be, for example, used to implement enforcement.
```

from the `tetra` CLI should be similar to this:

```json
{
  "process_kprobe": {
    "process": {
      "exec_id": "OjEwMTgzOTUyNjg1NjcwNDoyNjIzNw==",
      "pid": 26237,
      "uid": 1000,
      "cwd": "/home/ubuntu",
      "binary": "/usr/bin/cat",
      "arguments": "tracing_policy.yaml",
      "flags": "execve clone",
      "start_time": "2023-04-06T18:30:03.405730761Z",
      "auid": 1000,
      "parent_exec_id": "OjEwMTczMDEyNTQ3MjUxMDoyNjIwOA==",
      "refcnt": 1
    },
    "parent": {
      "exec_id": "OjEwMTczMDEyNTQ3MjUxMDoyNjIwOA==",
      "pid": 26208,
      "uid": 1000,
      "cwd": "/home/ubuntu",
      "binary": "/bin/bash",
      "flags": "execve clone",
      "start_time": "2023-04-06T18:28:14.004347210Z",
      "auid": 1000,
      "parent_exec_id": "OjEwMTczMDEwMjc2NTg0NjoyNjIwNw==",
      "refcnt": 2
    },
    "function_name": "__x64_sys_openat",
    "args": [
      {
        "int_arg": -100
      },
      {
        "string_arg": "tracing_policy.yaml"
      },
      {
        "int_arg": 0
      }
    ],
    "action": "KPROBE_ACTION_POST"
  },
  "time": "2023-04-06T18:30:03.406205829Z"
}
```

> You can see that we collected information on the process itself, with for
> example, the uid, the binary name, and the start\_time. But we also collected
> information on the parent of this process, here the binary `/bin/bash`. This is
> the same kind of information we get from the base process lifecycle sensors.
> Most importantly, we can see that we hooked the function `__x64_sys_openat`
> (for an arm64 host it would be `__arm64_sys_openat`) and we also collected the
> argument's value.

Reviewer: ```suggestion
You can see that Tetragon collected information on the process itself, with for
example, the uid, the binary name, and the start\_time, as well as on the process parent,
`/bin/bash` in this case. This is
the same kind of information we get from the base process lifecycle sensors.
Finally, we can see that we hooked the function `__x64_sys_openat`
(for an arm64 host it would be `__arm64_sys_openat`) and we also collected the
argument's value.
```

We can use this command to extract only the second arguments for the hooked
function, and run again `cat tracing_policy.yaml` on the side:

```shell
tetra getevents | jq 'select(has("process_kprobe")) | .process_kprobe.args[1].string_arg'
```

The output should be similar to this:
```
"/etc/ld.so.cache"
"/lib/x86_64-linux-gnu/libc.so.6"
"tracing_policy.yaml"
```

We can see that during the execution of `cat`, because it is dynamically
linked, it used the `openat` syscall to open the cache file for shared
libraries `ld.so.cache`, then the `libc.so.6` library and finally the file we
wanted to actually read `tracing_policy.yaml`. This is the kind of information
we could obtain by executing the program with `strace` for example, but here
Tetragon obtains this information directly from the kernel, transparently for
the executed program.

### Enforcement with TracingPolicy

Let's see how we can modify the previous TracingPolicy to do enforcement with
Tetragon.

{{< warning >}}
This example is just an illustration and should not be used in a production

> environment, filtering an argument for security can be difficult, especially

Reviewer: ```suggestion
environment.  There are many challenges in building a production-grade policy, that go beyond the scope
of this guide. For example, filtering an argument for security can be difficult, especially
```
